pilot/pkg/networking/core/envoyfilter/cluster_patch.go

```diff
@@ -182,6 +186,27 @@ func clusterMatch(cluster *cluster.Cluster, cp *model.EnvoyFilterConfigPatchWrap
 	return true
 }
 
+func waypointMatched(cluster *cluster.Cluster, cp *model.EnvoyFilterConfigPatchWrapper) bool {
+	_, _, hostname, port := model.ParseSubsetKey(cluster.Name)
+	// match all
+	if len(cp.Hostnames) > 0 {
+		if !hostContains(cp.Hostnames, hostname) {
+			return false
+		}
+	}
+
+	if cp.Match.GetWaypoint() == nil {
+		return true
```

Comment:
I don't see why it would be a waypoint match if we're not in a waypoint patch context
```suggestion
		return false
```